File: scala/src/virtual/abs/AbsVertex.scala
package virtual.abs

import scala.collection.mutable.HashSet

class AbsVertex(private var label: Integer) {

  type G <: AbsGraph
  type V <: AbsVertex
  type E <: AbsEdge

  def unbindGraph(): Unit = {
    incidentEdges.foreach(e => graph.removeEdge(e));

    this.graph = null.asInstanceOf[G]
  }

  def getGraph: G = graph

  def setGraph(g: G): Unit = {
    if (graph != null)
      unbindGraph()

    this.graph = g
  }

  def getIncidentEdges: HashSet[E] = incidentEdges

  override def toString: String = label.toString()

  private var incidentEdges: HashSet[E] = new HashSet[E]
  private var graph: G = _
}

File: cpp/generics/abs/AbsVertex.cpp
/*
 * AbsVertex.cpp
 *
 *  Created on: 19 mai 2013
 *      Author: Abdelhamid
 */

#include "AbsVertex.h"

template <class G, class V, class E>
AbsVertex<G, V, E>::AbsVertex(int label)
{
	this->label = label;
	this->incidentEdges = new set<E*>();
	this->graph = 0;
}

template <class G, class V, class E>
virtual void AbsVertex<G, V, E>::unbindGraph()
{
	set<E*>::iterator it;
	for(it=incidentEdges->begin();
		it!=incidentEdges->end();
		it++)
		this->graph->removeEdge(it.)


	std::for_each(incidentEdges->begin(), incidentEdges->end(), )
}
